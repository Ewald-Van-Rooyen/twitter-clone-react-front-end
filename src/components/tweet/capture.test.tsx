import TestRenderer from "react-test-renderer";
import ReactTestUtils from "react-dom/test-utils";
import ReactDOM from 'react-dom';
import {act} from 'react-dom/test-utils';

import React from "react";
import Capture from "../capture/capture";

let container: any;
const defaultProfilePic = "https://avatarfiles.alphacoders.com/174/174225.jpg";

describe("Capture component tests", () => {

    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    it("should render the Capture component", () => {
        const testRenderer = TestRenderer.create(
            <Capture/>
        );
        expect(testRenderer).toMatchSnapshot();
    });

    it("should match the props", () => {
        const testRenderer = TestRenderer.create(
            <Capture profilePic={defaultProfilePic}/>
        );
        const testInstance = testRenderer.root;

        expect(testInstance.props.profilePic).toBe(defaultProfilePic);

    });

    it("should check for default img url", () => {
        const testRenderer = TestRenderer.create(
            <Capture/>
        );
        const testInstance = testRenderer.root;
        const img = testInstance.findByType("img");

        expect(img.props.src).toEqual(defaultProfilePic);
        expect(img.props.className).toEqual("avatar");

    });

    it("should submit a tweet", () => {
        const newTweet = "giraffe are cool";
        act(() => {
            ReactDOM.render(<Capture/>, container);
        });

        const input = container.querySelector("textarea");

        expect(input.value).toEqual("");

        input.value = newTweet;
        ReactTestUtils.Simulate.change(input);
        expect(input.value).toEqual(newTweet);
    });
});
