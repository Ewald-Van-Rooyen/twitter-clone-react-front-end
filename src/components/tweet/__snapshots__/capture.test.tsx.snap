// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Capture component tests should render the Capture component 1`] = `
<div
  className="container"
>
  <div
    className="horizontal-container"
  >
    <img
      alt=""
      className="avatar"
      src="https://avatarfiles.alphacoders.com/174/174225.jpg"
    />
    <form
      className="tweet-form"
      onSubmit={[Function]}
    >
      <div
        className="capture-input-container"
      >
        <textarea
          className="capture-input"
          onChange={[Function]}
          placeholder="What's happening?"
          value=""
        />
      </div>
      <div
        className="tweet-button-container"
      >
        <button
          className="tweet-button twitter-blue-background"
          type="submit"
        >
          <span
            className="text"
          >
            Tweet
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;
